components/patient-access-rules/ui/src/main/resources/PhenoTips/PatientLock.xml

```diff
@@ -0,0 +1,397 @@
+      </content>
+      <extensionPointId>
+        <disabled>0</disabled>
+        <name>extensionPointId</name>
+        <number>1</number>
+        <prettyName>Extension Point ID</prettyName>
+        <size>30</size>
+        <unmodifiable>0</unmodifiable>
+        <classType>com.xpn.xwiki.objects.classes.StringClass</classType>
+      </extensionPointId>
+      <name>
+        <disabled>0</disabled>
+        <name>name</name>
+        <number>2</number>
+        <prettyName>Extension ID</prettyName>
+        <size>30</size>
+        <unmodifiable>0</unmodifiable>
+        <classType>com.xpn.xwiki.objects.classes.StringClass</classType>
+      </name>
+      <parameters>
+        <disabled>0</disabled>
+        <name>parameters</name>
+        <number>4</number>
+        <prettyName>Extension Parameters</prettyName>
+        <rows>10</rows>
+        <size>40</size>
+        <unmodifiable>0</unmodifiable>
+        <classType>com.xpn.xwiki.objects.classes.TextAreaClass</classType>
+      </parameters>
+      <scope>
+        <cache>0</cache>
+        <disabled>0</disabled>
+        <displayType>select</displayType>
+        <multiSelect>0</multiSelect>
+        <name>scope</name>
+        <number>5</number>
+        <prettyName>Extension Scope</prettyName>
+        <relationalStorage>0</relationalStorage>
+        <separator> </separator>
+        <separators>|, </separators>
+        <size>1</size>
+        <unmodifiable>0</unmodifiable>
+        <values>wiki=Current Wiki|user=Current User|global=Global</values>
+        <classType>com.xpn.xwiki.objects.classes.StaticListClass</classType>
+      </scope>
+    </class>
+    <property>
+      <content>{{velocity}}
+$xwiki.jsx.use('PhenoTips.PatientLock')##
+$xwiki.ssx.use('PhenoTips.PatientLock')##
+#set ($lockDocName = 'PhenoTips.PatientLock')
+#set ($lockObject = $doc.getObject($lockDocName))
+#set ($patient = $services.patients.getPatientById($doc.documentReference))
+#set ($patientAccess = $services.permissions.getPatientAccess($patient))
+#set ($patientLockingService = $xwiki.getDocument($lockDocName))
+#set ($lockingServiceRequestParams = "patientId=" + $patient.getId())
+{{html}}
+##Macro used to create help button.
+#macro (_createLockButton $link $icon $text)&lt;span class='buttonwrapper'&gt;&lt;a id="record-locking-button" class="button" href="${link}"&gt;&lt;span class = "fa fa-${icon}"&gt;&lt;/span&gt; ${text}&lt;/a&gt; &lt;span class="fa fa-question-circle xHelpButton" title="$services.localization.render('phenotips.patientLock.action.hint')"&gt;&lt;/span&gt;&lt;/span&gt;#end
+&lt;div class="locking-container"&gt;
+#if ("$!{lockObject}" == '')
+  #if ($patientAccess.hasAccessLevel('manage'))
+    #set ($lockingServiceRequestParams = $lockingServiceRequestParams + "&amp;lock=true")
+    #_createLockButton("$patientLockingService.getURL('view', $lockingServiceRequestParams)", 'lock', "$services.localization.render('phenotips.patientLock.action.lock')")
```

Comment:
Instead of leading out to a different document, I would use this as a sheet for the patient record:

    $doc.getURL('view', "sheet=${lockDocName}")